src/lib/y2storage/storage_manager.rb

```diff
@@ -387,6 +389,10 @@ def apply_storage_defaults
     #
     # @param [Devicegraph] devicegraph to copy
     def copy_to_staging(devicegraph)
+      if devicegraph.nil?
+        log.error("Not copying nil devicegraph to staging")
+        return
```

Comment:
I have not gone through the whole PR yet (tomorrow maybe), but this looks to me like a way to silent a rather serious error... so it can shoot us by the back when less expected.

If someones tries to assign a nil devicegraph we used to get an exception, that's how the bug was found... and the cause is very clear. We can easily know that some particular piece of code failed to calculate a devicegraph and it didn't handle that case.

But if we simply do nothing when nil is assigned to staging and we simply keep the previous outdated staging devicegraph, then we are producing a rather subtle bug. That can result in a completely different error at another future point of the installation... quite difficult to track and maybe even unnoticed for quite some time.

You can say "the show must go on"... but I honestly don't want the show to go in with the wrong staging devicegraph. :see_no_evil:

Reply:
I can do that check also on the application level, but I can almost guarantee you that we will get a very similar backtrace very soon. This code is much too fragile IMHO. And for the user, this means the installation stops right here and now.